src/map/skill.cpp

```diff
@@ -7630,6 +7630,29 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 
 		break;
 
+	case NPC_MOVE_COORDINATE:
+		{
+			clif_skill_nodamage(src, bl, skill_id, skill_lv, 1);
+			clif_skill_damage(src, bl, tick, status_get_amotion(src), 0, -30000, 1, skill_id, skill_lv, DMG_SINGLE);
+
+			int16 px = bl->x, py = bl->y;
+			if (!skill_check_unit_movepos(0, bl, src->x, src->y, 1, 1)) {
+				return 0;
+			}
+
+			clif_blown(bl);
+
+			// If caster is not a boss, switch coordinates with the target
+			if (status_get_class_(src) != CLASS_BOSS) {
+				if (!skill_check_unit_movepos(0, src, px, py, 1, 1)) {
```

Comment:
```suggestion
				if (!skill_check_unit_movepos(0, src, bl->x, bl->y, 1, 1)) {
```

Reply:
Thats not possible, because bl was already moved at this point in time!

Reply:
Reviewing from mobile so the "check" in the function name confused me if it actually moved the object or not!